domenetjenester/registerinnhenting/src/main/java/no/nav/ung/sak/domene/registerinnhenting/InntektAbonnentKlient.java

```diff
@@ -0,0 +1,140 @@
+
+    @Inject
+    public InntektAbonnentKlient(
+        @KonfigVerdi(value = "inntektskomponenten.url",
+            defaultVerdi = "http://ikomp.team-inntekt") String baseUrl,
+        OidcRestClient oidcRestClient) {
+        this.oidcRestClient = oidcRestClient;
+        this.opprettAbonnementURI = tilUri(baseUrl, "rs/api/v1/abonnement");
+        this.avsluttAbonnementURI = tilUri(baseUrl, "rs/api/v1/abonnement/%s");
+        this.hendelseStartURI = tilUri(baseUrl, "rest/v2/abonnement/hendelse/start");
+        this.hendelseURI = tilUri(baseUrl, "rest/v2/abonnement/hendelse");
+    }
+
+    public InntektAbonnement opprettAbonnement(AktørId aktørId, String formaal, List<String> filter,
+                                               String fomMaanedObservasjon, String tomMaanedObservasjon,
+                                               LocalDate sisteBruksdag) {
+        var request = new OpprettAbonnementRequest(
+            aktørId.getId(),
+            formaal,
+            filter,
+            fomMaanedObservasjon,
+            tomMaanedObservasjon,
+            sisteBruksdag
+        );
+        AbonnementResponse response;
+        try {
+            response = oidcRestClient.post(opprettAbonnementURI, request, AbonnementResponse.class);
+        } catch (Exception e) {
+            throw new TekniskException("UNG-947528", "Feil ved opprettelse av abonnement", e);
+        }
+
+        var abonnement = new InntektAbonnement(String.valueOf(response.abonnementId()), aktørId);
+        log.info("Opprettet abonnementId: {} for aktør: {}", response.abonnementId(), aktørId.getId());
+        return abonnement;
+    }
+
+    public long hentStartSekvensnummer(LocalDate dato) {
+        var request = new AbonnementHendelseStartApiInn(dato);
+        AbonnementHendelseStartApiUt response;
+        try {
+            response = oidcRestClient.post(hendelseStartURI, request, AbonnementHendelseStartApiUt.class);
+        } catch (Exception e) {
+            throw new TekniskException("UNG-440600", "Feil ved henting av startsekvensnummer", e);
+        }
+        return response.sekvensnummer();
+    }
+
+    public List<AbonnementHendelse> hentAbonnentHendelser(long fraSekvensnummer, List<String> filter) {
+        var request = new InntektHendelserRequest(fraSekvensnummer, INNTEKT_HENDELSE_LIMIT, filter);
+        AbonnementHendelseApiUt response;
+        try {
+            response = oidcRestClient.post(hendelseURI, request, AbonnementHendelseApiUt.class);
+        } catch (Exception e) {
+            throw new TekniskException("UNG-769025", "Feil ved henting av nye hendelser", e);
+        }
+        return response.data();
+    }
+
+    public void avsluttAbonnement(long abonnementId) {
+        try {
+            var uri = URI.create(String.format(avsluttAbonnementURI.toString(), abonnementId));
+            oidcRestClient.delete(uri);
+            log.info("Avslutter abonnement med id {}", abonnementId);
```

Comment:
Her er det to info-meldingen avslutter og avsluttet som begge logges etter at abonnementet er avsluttet, det bør holde med en av disse